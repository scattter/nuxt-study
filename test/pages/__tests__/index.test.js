import { shallowMount } from "@vue/test-utils";
import App from '@/pages/index';
import { getData } from "~/service/index.api";

// mock a file
jest.mock('~/service/index.api')

// mock axios
// jest.mock('axios')
// axios.get.mockResolvedValue({data: [{name: 'demo'}]})

describe('test for app', () => {

  it('should correctly render page', () => {
    const wrapper = shallowMount(App);
    expect(wrapper.html()).toMatchSnapshot();
  });

  // you can use mocks to mount App by options when mock current file's function
  // also you can user up axios to mock data
  // it('should correctly call api when render this page', (done) => {
  //   const mockApi = jest.fn(() => Promise.resolve({data: [{name: 'demo'}]}))
  //   const wrapper = shallowMount(App, { mocks: { $axios: { get: mockApi } } })
  //   setTimeout(() => {
  //     expect(mockApi).toBeCalledWith('http://localhost:8000/menu/tree')
  //     expect(wrapper.vm.nodeData).toEqual('demo')
  //     done()
  //   })
  // });

  // mock a file
  it('should correctly call api when render this page', (done) => {
    getData.mockImplementation(() => 'demo')
    const wrapper = shallowMount(App)
    setTimeout(() => {
      expect(getData).toHaveBeenCalled()
      expect(wrapper.vm.nodeData).toEqual('demo')
      done()
    })
  });
})
