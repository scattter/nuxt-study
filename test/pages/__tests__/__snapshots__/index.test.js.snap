// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for app should correct render 1`] = `
"<section>
  <h2>this is a demo</h2> <span></span>
  <h3>footer</h3>
</section>"
`;
